File: fetch.py
# coding: utf-8

from urllib2 import *
import re

def download_img(name, url):
    if url[-4:] in ('.jpg', '.gif', '.png'):
	pathname = './imgs/' + name + url[-4:]
    else:
	pathname = './imgs/' + name + '.jpg'
    try:
	f = open(pathname, 'w+')
	f.write(urlopen(url).read())
	f.close()
    except URLError, msg:
	print '!!!!!!!!!! DOWNLOAD IMAGE ERROR !!!!!!!!!!!'
	print msg
	print 'fail:', url
	return 
    print 'done:', pathname

def fetch_next_url(content):
    p = re.compile(r'<span class="current-comment-page">\[(.*?)\]</span>')
    m = p.search(content)
    if not m:
	return m
    page_number = int(m.groups()[0])-1
    return 'http://jandan.net/ooxx/page-%d' % (page_number)

def fetch_img(url):
    content = urlopen(url).read()
    content = content.replace('\r\n', '')
    content = content.replace('\n', '')

    commentlist = re.compile(r'''<ol class="commentlist"(.*)</ol>''')
    m = commentlist.search(content)
    if not m:
	return

    tmp = m.group(0)
    p = re.compile(r'''<li id="comment-(.*?)</li>''')
    lis = p.findall(tmp)

    if not lis:
	return

    p1 = re.compile(r'(.*?)">')
    p2 = re.compile(r'<p>.*?<img (.*?) ?/>')
    for g in lis:
	try:
	    id_ =  p1.search(g).groups()[0]
	
	    imgs = p2.search(g).groups()[0]
	    p3 = re.compile(r'src="(.*?)"')
	    imgs = p3.findall(imgs)
	    idx = 0
	    for img in imgs:
		download_img(id_+'_'+str(idx), img)
		idx+=1
	except AttributeError, msg:
	    print '!!!!!!!!!!!!!OCCUR ERROR!!!!!!!!!!!!!!!'
	    print msg
	    print g
	    sys.exit()

    return fetch_next_url(content)


url = 'http://jandan.net/ooxx/page-900'

for i in xrange(20):
    print url
    url = fetch_img(url)

File: fetch.lua


http = require("socket.http")
local gfind = string.gfind

function _download_img( img_url, filename )
   body, ret = http.request(img_url)
   io.open(filename, 'wb'):write(body)
   print(filename .. ' done')
end

function _parse_li(li)
   -- get img id
   local pattern = '%-(.-)">'
   local id = 0
   for _1, _ in gfind(li, pattern) do
      id = _1
      break
   end

   -- get img list
   local pattern = '<img src="(.-)" />'
   local idx = 1
   for img, _ in gfind(li, pattern) do
      local filename = id .. tostring(idx) .. string.sub(img, -4)
      _download_img(img, filename)
      idx = idx + 1
   end
end

function _parse_li_list(commentlist)
   local pattern = '<li id="comment(.-)</li>'
   for li, _ in gfind(commentlist, pattern) do
      _parse_li(li)
      -- print(li)
   end
end

function _parse_commentlist(cnt)
   local pattern = '<ol class="commentlist"(.-)</ol>'
   for commentlist, _ in gfind(cnt, pattern) do
      _parse_li_list(commentlist)
   end
end

function get_next_url( content )
   local pattern = '<span class="current%-comment%-page">%[(.-)%]</span>'
   local cur = ''
   for _1, _ in gfind(content, pattern) do
      cur = tonumber(_1) - 1
      break
   end

   local url = 'http://jandan.net/ooxx/page-' .. tostring(cur)
   return url
end

function download_ooxx_img(url)
   body, ret = http.request(url)
   if not body then
      return
   end

   local url = get_next_url(body)
   _parse_commentlist(body)
   return url
end

function main( url )
   local u = url
   for i = 1, 10 do
      print(u)
      u = download_ooxx_img(u)
   end
end

main('http://jandan.net/ooxx')